src/types/qwcompositor.h

```diff
@@ -93,6 +93,8 @@ class QW_EXPORT QWSurface : public QObject, public QWObject
     void new_subsurface();
 #if WLR_VERSION_MINOR > 16
     void precommit(const wlr_surface_state *state);
+    void map();
```

Comment:
Provide the signals on all versions. If the wlroots version <= 16, then emit the signals in QWXdgShellSurface